src/cascadia/TerminalSettingsModel/TerminalSettings.cpp

```diff
@@ -341,6 +346,8 @@ namespace winrt::Microsoft::Terminal::Settings::Model::implementation
         _RepositionCursorWithMouse = profile.RepositionCursorWithMouse();
 
         _ReloadEnvironmentVariables = profile.ReloadEnvironmentVariables();
+
+        _CloseOnExit = profile.CloseOnExit();

```

Comment:
this seems... weird. This should already be plumbed through to the `TermControl` somewhere else...

Reply:
Ah, right. It's not actually used by the `TermControl`. The pane itself handles that, in this blob of code:
https://github.com/microsoft/terminal/blob/9c16c5ca82943c2c466fb16a093a559d53a2fb0c/src/cascadia/TerminalApp/TerminalPaneContent.cpp#L240-L256

Looks like that's specifically looking up the CloseOnExit value out of the profile, not the `TerminalSettings`. I bet you could update that to use the value from the TerminalSettings too. 

This code has an example in that file of getting settings from the cache:
https://github.com/microsoft/terminal/blob/9c16c5ca82943c2c466fb16a093a559d53a2fb0c/src/cascadia/TerminalApp/TerminalPaneContent.cpp#L345

That'll get you a `TerminalSettingsCreateResult`, who's `.DefaultSettings()` will have the value you're looking for